src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -437,42 +682,129 @@ void EIRPlantLoopHeatPump::sizeLoadSide(EnergyPlusData &state)
+                                rho = compRho;
+                                tmpLoadVolFlow = companionVolFlowRate;
+                            }
+                        }
+                    }
+                }
+            } else {
+                tmpCapacity = Cp * rho * deltaT * tmpLoadVolFlow * this->heatSizingRatio;
             }
-            tmpCapacity = Cp * rho * state.dataSize->PlantSizData(pltLoadSizNum).DeltaT * tmpLoadVolFlow;
         } else if (this->companionHeatPumpCoil && this->companionHeatPumpCoil->loadSideDesignVolFlowRate > 0.0) {
             tmpLoadVolFlow = this->companionHeatPumpCoil->loadSideDesignVolFlowRate;
-            tmpCapacity = Cp * rho * state.dataSize->PlantSizData(pltLoadSizNum).DeltaT * tmpLoadVolFlow;
+            if (this->companionHeatPumpCoil->referenceCapacity == DataSizing::AutoSize) {
+                // use reverse init temp, e.g., if this is cooling use HWInitConvTemp
+                Real64 compLoadSideInitTemp =
+                    (this->EIRHPType == DataPlant::PlantEquipmentType::HeatPumpEIRCooling) ? Constant::HWInitConvTemp : Constant::CWInitConvTemp;
+                int compLoopNum = this->companionHeatPumpCoil->loadSidePlantLoc.loopNum;
+                if (compLoopNum > 0) {
+                    Real64 const compRho = FluidProperties::GetDensityGlycol(state,
+                                                                             state.dataPlnt->PlantLoop(compLoopNum).FluidName,
+                                                                             compLoadSideInitTemp,
+                                                                             state.dataPlnt->PlantLoop(compLoopNum).FluidIndex,
+                                                                             "EIRPlantLoopHeatPump::size()");
+                    Real64 const compCp = FluidProperties::GetSpecificHeatGlycol(state,
+                                                                                 state.dataPlnt->PlantLoop(compLoopNum).FluidName,
+                                                                                 Constant::CWInitConvTemp,
+                                                                                 state.dataPlnt->PlantLoop(compLoopNum).FluidIndex,
+                                                                                 "EIRPlantLoopHeatPump::size()");
+                    rho = compRho;
+                    Cp = compCp;
+                }
+                tmpCapacity = Cp * rho * state.dataSize->PlantSizData(pltLoadSizNum).DeltaT * tmpLoadVolFlow * this->heatSizingRatio;
+            } else {
+                tmpCapacity = this->companionHeatPumpCoil->referenceCapacity;
+            }
         } else {
             if (this->referenceCapacityWasAutoSized) tmpCapacity = 0.0;
             if (this->loadSideDesignVolFlowRateWasAutoSized) tmpLoadVolFlow = 0.0;
         }
```

Comment:
next is heat recovery HP connected to plant inlet/outlet used plant flow rate, and save temp capacity to use during subsequent sizing iterations